ci/build_setup.sh

```diff
@@ -14,23 +14,58 @@ NUM_CPUS=`grep -c ^processor /proc/cpuinfo`
 
 if [[ "$1" == bazel* ]]
 then
+  export BUILD_DIR=/build
+  # Make sure that "docker run" has a -v bind mount for /build, since cmake
+  # users will only have a bind mount for /source.
+  if [[ ! -a "${BUILD_DIR}" ]]
+  then
+    echo "${BUILD_DIR} mount missing - did you forget -v <something>:${BUILD_DIR}?"
+    exit 1
+  fi
+  export ENVOY_SRCDIR=/source
+  export ENVOY_CONSUMER_SRCDIR="${BUILD_DIR}/envoy-consumer"
+
+  # Make sure that /source doesn't contain /build on the underlying host
+  # filesystem, including via hard links or symlinks. We can get into weird
+  # loops with Bazel symlinking and gcovr's path traversal if this is true, so
+  # best to keep /source and /build in distinct directories on the host
+  # filesystem.
+  SENTINEL="${BUILD_DIR}"/bazel.sentinel
+  touch "${SENTINEL}"
+  if [[ -n "$(find -L "${ENVOY_SRCDIR}" -name "$(basename "${SENTINEL}")")" ]]
+  then
+    rm -f "${SENTINEL}"
+    echo "/source mount must not contain /build mount"
+    exit 1
+  fi
+  rm -f "${SENTINEL}"
+
+  # Environment setup.
   export USER=bazel
-  # Prefixing with bazel- is necessary to prevent creating symlink loops that
-  # confuses gcovr.
-  export TEST_TMPDIR=/source/bazel-build
-  export SRCDIR="$(realpath "${PWD}")"
+  export TEST_TMPDIR=/build/tmp
   export BAZEL="bazel"
   export BAZEL_COVERAGE="bazel-coverage"
   # Not sandboxing, since non-privileged Docker can't do nested namespaces.
-  export BAZEL_BUILD_OPTIONS="--strategy=CppCompile=standalone --strategy=CppLink=standalone \
-    --strategy=TestRunner=standalone --strategy=ProtoCompile=standalone \
-    --strategy=Genrule=standalone --verbose_failures --package_path %workspace%:.."
+  export BAZEL_BUILD_OPTIONS="--strategy=Genrule=standalone --spawn_strategy=standalone \
```

Comment:
We're building for non-consuming now in `/build/envoy`, this is where we invoke Bazel. So, it needs to know that it can still find the Envoy source in `/source` https://bazel.build/versions/master/docs/bazel-user-manual.html#flag--package_path